src/devenv.nix

```diff
@@ -124,20 +126,33 @@ pkgs.writeScriptBin "devenv" ''
       echo $spec
     
       # copy container
-      if [[ ''${subcommand[--copy]} != false || ''${subcommand[--docker-run]} != false ]]; then
+      if [[ ''${subcommand[--copy]} != false || ''${subcommand[--docker-run]} != false || ''${subcommand[--podman-run]} != false ]]; then
         copyScript=$($CUSTOM_NIX/bin/nix $NIX_FLAGS build --print-out-paths --no-link --impure ".#devenv.containers.\"$container\".copyScript")
 
+        args=()
+
         if [[ ''${subcommand[--docker-run]} == true ]]; then
-          registry=docker-daemon:
+          registry=local-docker
+        elif [[ ''${subcommand[--podman-run]} == true ]]; then
+          registry=local
         else
           registry="''${subcommand[--registry]}"
         fi
-        $copyScript $spec $registry ''${subcommand[--copy-args]}
+        if [[ -n "''${subcommand[--image]}" ]]; then
+          args+=(--image "''${subcommand[--image]}")
+        fi
+
+        $copyScript "$spec" --registry "$registry" "''${args[@]}" ''${subcommand[--copy-args]}
       fi
 
-      # docker run
       if [[ ''${subcommand[--docker-run]} != false ]]; then
-        $($CUSTOM_NIX/bin/nix $NIX_FLAGS build --print-out-paths --no-link --impure ".#devenv.containers.\"$container\".dockerRun")
+        # shellcheck disable=SC1090
+        source "$(command -v docopts.sh)"
+        $($CUSTOM_NIX/bin/nix $NIX_FLAGS build --print-out-paths --no-link --impure ".#devenv.containers.\"$container\".dockerRun") -- $(docopt_get_values subcommand '<run-args>')
+      if [[ ''${subcommand[--podman-run]} != false ]]; then
```

Comment:
there needs to be a 'fi' on the preceding line, or the if needs to be an 'elif'.